cql3/statements/function_statement.cc

```diff
@@ -27,6 +28,45 @@ future<> create_function_statement_base::check_access(query_processor& qp, const
     }
 }
 
+future<> drop_function_statement_base::check_access(query_processor& qp, const service::client_state& state) const
+{
+    create_arg_types(qp);
+    shared_ptr<functions::function> func;
+    if (_args_present) {
+        func = functions::functions::find(_name, _arg_types);
+        if (!func) {
+            return make_ready_future<>();
+        }
+    } else {
+        auto funcs = functions::functions::find(_name);
+        if (!funcs.empty()) {
+            auto b = funcs.begin();
+            auto i = b;
+            if (++i != funcs.end()) {
+                return make_ready_future<>();
+            }
+            func = b->second;
```

Comment:
We're checking here if the range returned by `functions::functions::find` is singular by comparing begin()++ with end(). If the range is not singular, the query will fail not because of lacking permissions, but because we won't know which function to drop.
I agree that we should have a test for that, I'm adding it in the rebase